File: tinkerpop3/src/test/java/io/shiftleft/overflowdb/testdomains/simple/SimpleDomain.java
package io.shiftleft.overflowdb.testdomains.simple;

import io.shiftleft.overflowdb.OdbConfig;
import io.shiftleft.overflowdb.OdbGraph;

import java.util.Arrays;

public class SimpleDomain {
  public static OdbGraph newGraph() {
    return newGraph(OdbConfig.withoutOverflow());
  }

  public static OdbGraph newGraph(OdbConfig config) {
    return OdbGraph.open(
        config,
        Arrays.asList(OdbTestNode.factory),
        Arrays.asList(OdbTestEdge.factory)
    );
  }
}

File: traversal/src/test/scala/io/shiftleft/overflowdb/traversal/testdomains/simple/SimpleDomain.scala
package io.shiftleft.overflowdb.traversal.testdomains.simple

import io.shiftleft.overflowdb.traversal.{Traversal, TraversalSource, help}
import io.shiftleft.overflowdb.{OdbConfig, OdbGraph}
import java.util

import io.shiftleft.overflowdb.traversal.help.{Doc, TraversalHelp}

/** probably the most simple domain one can think of:
 * Thing --- Connection --> Thing
 * */
object SimpleDomain {
  def newGraph: OdbGraph = newGraph(OdbConfig.withoutOverflow)

  def newGraph(config: OdbConfig): OdbGraph =
    OdbGraph.open(config,
      util.Arrays.asList(Thing.factory),
      util.Arrays.asList(Connection.factory))

  def traversal(graph: OdbGraph) = new SimpleDomainTraversalSource(graph)

  lazy val help = new TraversalHelp(getClass.getPackage.getName)
}

@help.TraversalSource
class SimpleDomainTraversalSource(graph: OdbGraph) extends TraversalSource(graph) {

  @Doc("all things")
  def things: Traversal[Thing] = label(Thing.Label).cast[Thing]

  lazy val help: String = SimpleDomain.help.forSources
}

